assets/js/modules/adsense/datastore/settings.js

```diff
@@ -0,0 +1,444 @@
+	 */
+	*submitChanges() {
+		yield {
+			payload: {},
+			type: START_SUBMIT_CHANGES,
+		};
+
+		const { error } = yield {
+			payload: {},
+			type: SUBMIT_CHANGES,
+		};
+
+		yield {
+			payload: {},
+			type: FINISH_SUBMIT_CHANGES,
+		};
+
+		return { error };
+	},
+
+	/**
+	 * Sets the accountSetupComplete flag to true and submits all changes.
+	 *
+	 * This asynchronous action is used to couple setting of the flag with the
+	 * submission of the data, to avoid getting the value out of sync.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {boolean} True on success, false on failure.
+	 */
+	*completeAccountSetup() {
+		const success = yield {
+			payload: {},
+			type: COMPLETE_ACCOUNT_SETUP,
+		};
+		return success;
+	},
+
+	/**
+	 * Sets the siteSetupComplete flag to true and submits all changes.
+	 *
+	 * This asynchronous action is used to couple setting of the flag with the
+	 * submission of the data, to avoid getting the value out of sync.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return {boolean} True on success, false on failure.
+	 */
+	*completeSiteSetup() {
+		const success = yield {
+			payload: {},
+			type: COMPLETE_SITE_SETUP,
+		};
+		return success;
+	},
+
+	receiveOriginalAccountStatus( originalAccountStatus ) {
+		invariant( originalAccountStatus, 'originalAccountStatus is required.' );
+
+		return {
+			payload: { originalAccountStatus },
+			type: RECEIVE_ORIGINAL_ACCOUNT_STATUS,
+		};
+	},
+};
+
+export const controls = {
+	[ FETCH_SAVE_USE_SNIPPET ]: ( { payload } ) => {
+		const { useSnippet } = payload;
+		return API.set( 'modules', 'adsense', 'use-snippet', { useSnippet } );
+	},
+	[ SUBMIT_CHANGES ]: createRegistryControl( ( registry ) => async () => {
+		// This action shouldn't be called if settings haven't changed,
+		// but this prevents errors in tests.
+		if ( registry.select( STORE_NAME ).haveSettingsChanged() ) {
+			const { error } = await registry.dispatch( STORE_NAME ).saveSettings();
+
+			if ( error ) {
+				return { error };
+			}
+		}
+
+		await API.invalidateCache( 'modules', 'adsense' );
+		return {};
+	} ),
+	[ COMPLETE_ACCOUNT_SETUP ]: createRegistryControl( ( registry ) => async () => {
+		await registry.dispatch( STORE_NAME ).setAccountSetupComplete( true );
+		// canSubmitChanges cannot be checked before here because the settings
+		// won't have changed yet.
+		if ( ! registry.select( STORE_NAME ).canSubmitChanges() ) {
+			// Unset flag again.
+			await registry.dispatch( STORE_NAME ).setAccountSetupComplete( false );
+			return false;
+		}
+		const { error } = await registry.dispatch( STORE_NAME ).submitChanges();
+		if ( error ) {
+			// Unset flag again.
+			await registry.dispatch( STORE_NAME ).setAccountSetupComplete( false );
+			return false;
+		}
+		return true;
+	} ),
+	[ COMPLETE_SITE_SETUP ]: createRegistryControl( ( registry ) => async () => {
+		await registry.dispatch( STORE_NAME ).setSiteSetupComplete( true );
```

Comment:
Both `accountSetupComplete` and `siteSetupComplete` are settings (flags) that need to be set to indicate the user has completed these two respective parts of the setup. In case saving these "complete" settings fails for some reason, the values should be set back to `false`, to avoid having our client functionality behave as if setup was complete when it actually isn't.